### PR TITLE
Add vesting schedule

### PR DESCRIPTION
**Pull request by hobbydev71**

### CODE REVIEW

File: dist/types/models/index.d.ts [hunk 1]
```diff
@@ -5,3 +5,4 @@ export { Call } from "./Call";
 export { Account } from "./Account";
 export { Transfer } from "./Transfer";
 export { SystemTokenTransfer } from "./SystemTokenTransfer";
+export { VestingSchedule } from "./VestingSchedule";
```
Comment (aliXsed) [inline dist/types/models/index.d.ts]: You may want to remove the generated files under dist or ignore that folder when committing

File: dist/types/models/index.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 "use strict";
 // SPDX-License-Identifier: Apache-2.0
 Object.defineProperty(exports, "__esModule", { value: true });
-exports.SystemTokenTransfer = exports.Transfer = exports.Account = exports.Call = exports.Event = exports.Extrinsic = exports.Block = void 0;
+exports.VestingSchedule = exports.SystemTokenTransfer = exports.Transfer = exports.Account = exports.Call = exports.Event = exports.Extrinsic = exports.Block = void 0;
```
Comment (aliXsed) [inline dist/types/models/index.js]: Same here, this file can be removed

File: schema.graphql [hunk 3]
```diff
@@ -89,3 +89,10 @@ type SystemTokenTransfer @entity {
   timestamp: Date @index
   extrinsic: Extrinsic
 }
+
+type VestingSchedule @entity {
+  id: ID!
+  signer: String @index
+  to: String
+  data: String
```
Comment (aliXsed) [inline schema.graphql]: Can you better define the type of the `data` field here? The structure that we have for the data part of a VestingSchedule transaction should look like:
```
{
  start: Int
  period: Int
  periodCount: Int
  PerPeriod: BigInt
}
```
You can possibly remove `data` and just have the above field, or you can define a type for that field as above.

Comment (hobbydev71) [reply schema.graphql]: Yeah. Currently I have used this field as a string. and exchange this field type on web-wallet.
I will change the field type now.

File: dist/types/models/index.js (hunk 2 above)
Comment (hobbydev71) [reply dist/types/models/index.js]: Will remove it now.

File: dist/types/models/index.d.ts (hunk 1 above)
Comment (hobbydev71) [reply dist/types/models/index.d.ts]: Got it.

File: .gitignore [hunk 4]
```diff
@@ -1,4 +1,6 @@
 node_modules
 
 /.data
-/dist
\ No newline at end of file
+
+dist/
+src/types
```
Comment (Douglasacost) [inline .gitignore:6]: this type folder is not necessary?

File: project.yaml [hunk 5]
```diff
@@ -6,11 +6,11 @@ repository: https://github.com/hobbydev71/subquery-nodle.git
...
   file: ./schema.graphql
 network:
-  endpoint: wss://node-6913072722034561024.lh.onfinality.io/ws?apikey=84d77e2e-3793-4785-8908-5096cffea77a
-  genesisHash: '0x0fd167da222f10dae826a0433fe4daadcf4a36c2a6bd5c1b085857d5b79030af'
+  endpoint: wss://nodle-parachain.api.onfinality.io/public-ws
+  genesisHash: '0x97da7ede98d7bad4e36b4d734b6055425a3be036da2a332ea5a7037656427a21'
 dataSources:
   - kind: substrate/Runtime
-    startBlock: 1
+    startBlock: 343031
```
Comment (Douglasacost) [inline project.yaml:13]: This change here is okay?

Comment (Douglasacost) [inline project.yaml:13]: This change here is okay?

**Review by Douglasacost — DISMISSED** [review]
Nice job. Just some questions 👍

File: .gitignore (hunk 4 above)
Comment (fastendev) [reply .gitignore:6]: Yeah that folder is generated by  schema graphql when execute codegen command.

Comment (hobbydev71) [reply .gitignore:6]: yeah, the type folder is generated by schema graphql when execute codgen command.

File: project.yaml (hunk 5 above)
Comment (hobbydev71) [reply project.yaml:13]: I tried to get fetch specific data quickly on database. So I have changed the startblock point.

Comment (hobbydev71) [reply project.yaml:13]: Okay

Comment (aliXsed) [reply project.yaml:13]: Now it's time to start it back from 1 for production

**Review by aliXsed — APPROVED** [review]
I just left a trivial comment to start the blocks from 1 but I understand while you are testing you need a quick start up. So if you are going to fix it later it's okay

**Comment by hobbydev71** [comment]
> I just left a trivial comment to start the blocks from 1 but I understand while you are testing you need a quick start up. So if you are going to fix it later it's okay

Thank you. I will check once more. and change the startblock to 1.

**Pull request merged**